tools/study-room-scraper/src/study-room-scraper.ts

```diff
@@ -0,0 +1,170 @@
+        if (paraText.includes("\n")) {
+          paraText = paraText.replaceAll("\n", ", ");
+          if (!paraText.endsWith(":")) {
+            paraText += ". ";
+          }
+        }
+        descriptionText += paraText + " ";
+      });
+      descriptionText = descriptionText.replace(/\s{2,}/g, " ").trim(); // Remove extra spaces
+      descriptionText = descriptionText.replace(/\s+,/g, ","); // Remove spaces before commas
+      descriptionText = descriptionText.replace(/\.\s*\./g, "."); // Remove extra periods
+      descriptionText = descriptionText.replace(".,", "."); // Remove commas after periods
+    } else {
+      // General processing for other rooms
+      const descriptionParts: string[] = [];
+      let combinedDescription = "";
+
+      descriptionHeader.contents().each((_, content) => {
+        if (content.nodeType === 3) {
+          const textContent = $(content).text().trim();
+          if (textContent) {
+            descriptionParts.push(textContent);
+          }
+        } else if (content.nodeType === 1) {
+          const child = $(content);
+          if (child.is("p, ul, li, strong, em, span, br")) {
+            if (child.is("ul")) {
+              child.find("li").each((_, li) => {
+                descriptionParts.push("- " + $(li).text().trim());
+              });
+            } else if (child.is("br")) {
+              descriptionParts.push("\n");
+            } else {
+              descriptionParts.push(child.text().trim());
+            }
+          }
+        }
+      });
+
+      // join parts and replace newline placeholders with commas
+      combinedDescription = descriptionParts.join(" ").replace(/\n+/g, ", ");
+
+      // clean up
+      combinedDescription = combinedDescription
+        .replace(/\s*,\s*/g, ", ")
+        .replace(/\s*\.\s*/g, ". ")
+        .replace(/\s{2,}/g, " ")
+        .replace(/\.,/g, ".")
+        .replace(/\.\s*\./g, ".");
+
+      // description ends with a single period
+      combinedDescription = combinedDescription.replace(/\.\s*$/, ".");
+
+      descriptionText = combinedDescription.trim();
+    }
+
+    if (descriptionText) {
+      room.description = descriptionText;
+      if (!room.description.endsWith(".")) {
+        room.description += ".";
+      }
+    }
+
+    logger.info(`Scraped Room ${RoomId}`, { room });
+    return room;
+  } catch (error) {
+    logger.error(`Error fetching room information for room ${RoomId}`, { error });
+    return room;
+  }
+}
+
+export async function scrapeStudyLocations(): Promise<StudyLocations> {
+  const date = new Date();
+  const start = date.toLocaleDateString("en-CA", {
+    year: "numeric",
+    month: "2-digit",
+    day: "2-digit",
+  });
+  date.setDate(date.getDate() + 3);
+  const end = date.toLocaleDateString("en-CA", {
+    year: "numeric",
+    month: "2-digit",
+    day: "2-digit",
+  });
+  const studyLocationsMap: StudyLocations = {};
+  const rids: Set<string> = new Set();
```

Comment:
You can type the set's construction.

```suggestion
  const rids = new Set<string>();
```